app/main.py

```diff
@@ -1,25 +1,65 @@
-from abc import ABC
+class IntegerRange:
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
 
+    def __set_name__(
+            self, owner: "SlideLimitationValidator",
+            name: str) -> None:
+        self.protected_name = f"_{name}"
 
-class IntegerRange:
-    pass
+    def __get__(
+            self, instance: "SlideLimitationValidator",
+            owner: "SlideLimitationValidator") -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self, instance: "SlideLimitationValidator",
+            value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("value should be int")
+        elif not self.min_amount <= value <= self.max_amount:
+            raise ValueError("value is out of range")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
-class SlideLimitationValidator(ABC):
-    pass
+class SlideLimitationValidator:
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(
+            self, name: str,
+            limitation_class: SlideLimitationValidator) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        try:
+            self.limitation_class(visitor.age, visitor.weight, visitor.height)
+            return True
+        except ValueError:
```

Comment:
You should catch here not only `ValueError`, but `TypeError` as well since it is declared